File: javascript/hamming/hamming.js
var compute = function(a, b) {
  var ctr = 0;
  for(var i = 0; i < a.length; i++) {
    if (a[i] != b[i]) {
      ctr = ctr += 1;
    }
  }
  return ctr;
};

module.exports.compute = compute;

File: scala/hamming/src/main/scala/hamming.scala
object Hamming {
  def compute(dna1: String, dna2: String): Int = {
    if (dna1.length != dna2.length) {
      throw new IllegalArgumentException
    }
    var ctr = 0
    for (i <- 0 until dna1.length) {
      if (dna1(i) != dna2(i)) {
        ctr += 1
      }
    }
    return ctr
  }
}
